### PR TITLE
[Aliases] Adding babel aliases

### PR DESCRIPTION
**Pull request by DLotero**
## Summary

-Changed the setup script to enable aliases.
-Changed template for using aliases.

### CODE REVIEW

File: generators/app/tasks/appSetup/babelrcSetup.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
 module.exports = function babelrcSetup() {
-  this.fs.extendJSON(`${this.projectName}/.babelrc`, { plugins: ['import-glob'] });
+  this.fs.extendJSON(`${this.projectName}/.babelrc`, {
+    plugins: [
+      'import-glob',
+      [
+        'module-resolver',
+        {
+          root: './src',
```
Comment (sfernandez11) [inline generators/app/tasks/appSetup/babelrcSetup.js]: I think this must be: `root: ['./src'],`

File: generators/app/tasks/installDependencies.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const runCommand = require('./runCommand');
 
 const DEPENDENCIES = [
   'apisauce',
+  'babel-plugin-module-resolver',
```
Comment (sfernandez11) [inline generators/app/tasks/installDependencies.js]: This is a `devDependencie`, move it to `DEV_DEPENDENCIES`

File: generators/app/templates/App.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
 import { Provider } from 'react-redux';
+import '@config';
+import store from '@redux/store';
```
Comment (sfernandez11) [inline generators/app/templates/App.js]: Remove empty line between global imports

File: generators/app/templates/src/app/components/AppNavigator/index.js [hunk 4]
```diff
@@ -4,9 +4,9 @@ import { BackHandler } from 'react-native';
 import { connect } from 'react-redux';
 import { NavigationActions } from 'react-navigation';
 import { createReduxContainer } from 'react-navigation-redux-helpers';
+import { ROOT } from '@constants/platform';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/AppNavigator/index.js]: Same, if you leave this space then a linter problem occurs.

File: generators/app/templates/src/app/components/CustomText/styles.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/CustomText/styles.js]: Same

File: generators/app/templates/src/app/components/CustomTextInput/styles.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/CustomTextInput/styles.js]: Same

File: generators/app/templates/src/app/components/Drawer/components/DrawerMenu/styles.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/Drawer/components/DrawerMenu/styles.js]: Same

File: generators/app/templates/src/app/components/Drawer/components/DrawerOverlay/styles.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/Drawer/components/DrawerOverlay/styles.js]: Same

File: generators/app/templates/src/app/components/Loadable/index.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { View, ActivityIndicator } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/Loadable/index.js]: Same

File: generators/app/templates/src/app/components/Loadable/styles.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/components/Loadable/styles.js]: Same

File: generators/app/templates/src/app/index.ejs [hunk 11]
```diff
@@ -1,30 +1,30 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/index.ejs]: Same

File: generators/app/templates/src/app/screens.ejs [hunk 12]
```diff
@@ -4,14 +4,14 @@
 
 import { <% if (features.tabs) { %>createBottomTabNavigator, <% } %>createStackNavigator } from 'react-navigation';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/screens.ejs]: Same

File: generators/app/templates/src/app/screens.ejs [hunk 13]
```diff
@@ -4,14 +4,14 @@
...
 
-import Routes from '../constants/routes';
-import { stackNavConfig, screensNavOptions <% if (features.tabs) { %>, tabNavConfig<% } %> } from '../config/navigation';
+import Routes from '@constants/routes';
+import { stackNavConfig, screensNavOptions <% if (features.tabs) { %>, tabNavConfig<% } %> } from '@config/navigation';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/screens.ejs]: Same

File: generators/app/templates/src/app/screens/Home/layout.ejs [hunk 14]
```diff
@@ -5,9 +5,9 @@ import PropTypes from 'prop-types';
 import { View } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/screens/Home/layout.ejs]: Same

File: generators/app/templates/src/app/screens/Home/styles.js [hunk 15]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/screens/Home/styles.js]: Same

File: generators/app/templates/src/app/screens/Login/styles.js [hunk 16]
```diff
@@ -1,6 +1,6 @@
 import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/app/screens/Login/styles.js]: Same

File: generators/app/templates/src/config/i18n.js [hunk 17]
```diff
@@ -1,8 +1,9 @@
 import I18n from 'react-native-i18n';
```
Comment (sfernandez11) [inline generators/app/templates/src/config/i18n.js]: Same

File: generators/app/templates/src/config/i18n.js [hunk 18]
```diff
@@ -1,8 +1,9 @@
...
 
 // eslint-disable-next-line import/no-unresolved, import/extensions
```
Comment (sfernandez11) [inline generators/app/templates/src/config/i18n.js]: Delete commented `eslint-disable-next-line` please

File: generators/app/templates/src/config/navigation.ejs [hunk 19]
```diff
@@ -3,10 +3,10 @@ import React from 'react';
 import { Button } from 'react-native';
```
Comment (sfernandez11) [inline generators/app/templates/src/config/navigation.ejs]: Same

File: generators/app/templates/src/redux/pushNotifications/actions.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 import { createTypes } from 'redux-recompose';
+import { isAndroid } from '@constants/platform';
```
Comment (sfernandez11) [inline generators/app/templates/src/redux/pushNotifications/actions.js]: Same

File: generators/app/templates/src/redux/pushNotifications/actions.js [hunk 21]
```diff
@@ -1,7 +1,7 @@
...
 
-import { isAndroid } from '../../constants/platform';
-import PushNotificationsService from '../../services/PushNotificationsService';
+import PushNotificationsService from '.@services/PushNotificationsService';
```
Comment (sfernandez11) [inline generators/app/templates/src/redux/pushNotifications/actions.js]: ⚠️ => `import PushNotificationsService from '@services/PushNotificationsService';`

File: generators/app/templates/src/redux/store.ejs [hunk 22]
```diff
@@ -7,7 +7,7 @@ import {
 } from 'react-navigation-redux-helpers';
 import { fetchMiddleware, configureMergeState } from 'redux-recompose';
```
Comment (sfernandez11) [inline generators/app/templates/src/redux/store.ejs]: Same

File: generators/app/tasks/appSetup/babelrcSetup.js (hunk 1 above)
Comment (DLotero) [reply generators/app/tasks/appSetup/babelrcSetup.js]: as far as I could find, it's the same thing!

File: generators/app/tasks/appSetup/babelrcSetup.js [hunk 23]
```diff
@@ -1,3 +1,23 @@
 module.exports = function babelrcSetup() {
-  this.fs.extendJSON(`${this.projectName}/.babelrc`, { plugins: ['import-glob'] });
+  this.fs.extendJSON(`${this.projectName}/.babelrc`, {
```
[the remark this answers is not in the document]
Comment (guidoprinc) [reply generators/app/tasks/appSetup/babelrcSetup.js]: i think that `.bablerc` was being replace with `babel.config.js`. I must check this

[the remark this answers is not in the document]
Comment (DLotero) [reply generators/app/tasks/appSetup/babelrcSetup.js]: https://babeljs.io/docs/en/configuration

> You want to programmatically create the configuration? babel.config.js is for you!

I think `babel.config.js` is the way to go. Also, the documentation explicitly recommends using it, so I think that's the way to go. @guidoprinc @sfernandez11  thoughts?

[the remark this answers is not in the document]
Comment (guidoprinc) [reply generators/app/tasks/appSetup/babelrcSetup.js]: according with https://babeljs.io/docs/en/configuration, babel.config.js is our best option

[the remark this answers is not in the document]
Comment (sfernandez11) [reply generators/app/tasks/appSetup/babelrcSetup.js]: Great! We have a card in trello for this change https://trello.com/c/XTVlq67q/96-migrar-babelrc-a-babelconfigjs. Do we make both changes in this PR or in a separate one (for change `bebelrc` for `babel.config.js`)?

[the remark this answers is not in the document]
Comment (WFolini) [reply generators/app/tasks/appSetup/babelrcSetup.js]: I let this for now, and in another PR will change that

File: generators/app/templates/src/services/PushNotificationsService.js [hunk 24]
```diff
@@ -1,12 +1,11 @@
 import { Platform, Alert } from 'react-native';
+import api from '@config/api'; // eslint-disable-line import/first
```
Comment (WFolini) [inline generators/app/templates/src/services/PushNotificationsService.js]: You can delete this comment now

**Pull request merged**